orte/util/nidmap.c

```diff
@@ -296,6 +289,13 @@ int orte_util_nidmap_create(opal_pointer_array_t *pool, char **regex)
                 }
                 continue;
             }
+            if ('.' == node[i]) {
+                /* just use the entire name */
+                fullname = true;
+                break;
+            }
+            /* this is either an alpha or '-' */
+            assert(isalpha(node[i]) || '-' == node[i]);
```

Comment:
Something doesn't look right here - why would you want to assert if the character is illegal? I should think we wouldn't want the daemon to simply crash. Let's instead do a show_help reporting the unsupported name and  return an error code.